src/services/userPermissions.js

```diff
@@ -0,0 +1,12 @@
+import {
+    fetchUserPermissionsData
+} from "./railcontent";
+
+let userPermissionsPromise = null;
+
+export async function fetchUserPermissions() {
+    if (!userPermissionsPromise) {
```

Comment:
My intention here is to ensure there is only one execution of this promise per request.  My understanding is on every request the dom is rebuilt in the browser, so this variable (userPermissionsPromise) will be cleared on every refresh.  @miguelfondeur is this correct or I'm I missing something in my understanding here.

I am not sure how this will behave on Mobile  @csillaj is this behavior similar?  Is there a DOM that's rebuilt every request will this persist somehow?

If not is there some other way to ensure we only call a promise once every request on both FE/MA?

Reply:
@robwillems React Native doesn't have a DOM in the same way as web has, it has a Virtual DOM. Every time a component changes, RN doesn't re-render the whole DOM, it searches where the change occurred and calculates the minimal changes needed. Also in RN DOM elements are translated to platform-specific (android, ios) UI components.
Something else I also might have to mention is that in the app we don't (can't) use this library as a script embedded in html, we import each method we are using directly from the library in our services.
So if this fetchUserPermissions is called only once in a method, than when we import and call the method (e.g.: fetchAll) fetchUserPermissions will only be called once.

Reply:
Thanks, @csillaj. That's what I needed to know.  I'll come up with something else for this.